ada/ast.py

```diff
@@ -94,6 +94,51 @@ class AdaNode(ASTNode):
         public=True
     )
 
+    @langkit_property(ignore_warn_on_node=True)
+    def withed_unit_helper(unit_name=T.Name):
+        """
+        Static method helper. Fetch the unit designated by unit_name. Return
+        its body.
+        """
+        spec = Var(Self.designated_compilation_unit(
```

Comment:
Minor: you could store `unit_name.as_symbol_array` in a `Var` given that all paths compute it 3 times in a row.